frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -218,413 +223,520 @@ export function DynamicConfiguration () {
-                                        checked={state.form.configuration.EnabledServices[service]}
-                                    >
-                                        { configurableServices[service] }
-                                    </Toggle>
-                                );
-                            })}
+        }
+        return output;
+    }
+
+    /**
+     * Converts a JSON object into an outline structure represented as React nodes.
+     *
+     * @param {object} [json={}] - The JSON object to be converted.
+     * @returns {React.ReactNode[]} - An array of React nodes representing the outline structure.
+     */
+    function jsonToOutline (json = {}) {
+        const output: React.ReactNode[] = [];
+
+        for (const key in json) {
+            const value = json[key];
+            output.push((<li><p><strong>{key}</strong></p></li>));
+
+            if (isObject(value)) {
+                const recursiveJson = jsonToOutline(value); // recursively call
+                output.push((recursiveJson));
+            }
+        }
+        return <ul>{output}</ul>;
+    }
+
+    const changesDiff = getJsonDifference(applicationConfig.configuration, state.form.configuration);
+
+    return (
+        <>
+            <Modal 
+                visible={modalVisible}
+                onDismiss={() => setModalVisible(false)}
+                header={<Header>Confirm changes</Header>}
+                footer={
+                    <Box float='right'>
+                        <SpaceBetween direction='horizontal' size='xs'>
+                            <Button onClick={() => setModalVisible(false)}>Cancel</Button>
+                            <Button 
+                                variant='primary'
+                                loading={state.formSubmitting}
+                                disabled={_.isEmpty(changesDiff)}
+                                onClick={async () => {
+                                    await handleSubmit();
+                                    setModalVisible(false);
+                                }
+                                }>Save</Button>
                         </SpaceBetween>
-                    </ContentLayout>
-                </ExpandableSection>
-                <ExpandableSection headerText='EMR Config' variant='default' defaultExpanded>
-                    <ExpandableSection 
-                        headerText='Applications'
-                        variant='default' 
-                        headingTagOverride='h3' 
-                        headerDescription='A list of applications for Amazon EMR to install and configure when launching the cluster.'
+                    </Box>
+                }
+            >
+                <SpaceBetween size={'s'}>
+                    <Container>
+                        <TextContent>
+                            {_.isEmpty(changesDiff) ? <p>No changes detected</p> : jsonToOutline(changesDiff)}
```

Comment:
Can we use this same logic to disable the "Save Changes" button so that we're only enabling it when the user has made changes?

Reply:
I figured this way the user would at least know why the button is deactivated.

Reply:
I'm talking about the Save changes button on line 671 where we determine if that button is disabled or not - can we apply this same logic there? So the button is disabled if:

* the form is submitting
* the form is invalid 
* no changes have yet been made

Reply:
if I disable that button, the user wont know why its disabled. I made it so that you could still click on it but then it shows the modal like this:

![Screenshot 2024-06-05 at 12 12 32 PM](https://github.com/awslabs/mlspace/assets/28429388/b2105d69-e446-48ba-b72d-241511b96154)

So the user will still not be allowed to submit without changes but will have a visual indication as to why it cant. Ultimately Ill change it to whatever you think makes more sense.

Reply:
You know what I like that better than what I was thinking, let's keep that